### PR TITLE
Add events for functions and executions

### PR DESCRIPTION
**Pull request by kodumbeats**
## What does this PR do?

This PR adds events for the following actions in the API:
- `functions.create`
- `functions.update`
- `functions.delete`
- `functions.tags.create`
- `functions.tags.update`
- `functions.tags.delete`
- `functions.executions.create`

It also adds the following event in the Functions worker:
- `functions.executions.update`

## Test Plan

- Create webhook for new events
- Trigger the events through the console
- Observe logs for `appwrite-worker-webhooks` for webhook triggers
- Observe webhook target endpoint (`request-catcher` in testing)

## Todo

- [X] Create events
- [x] Create webhook tests
- [x] Update changelog

## Related PRs and Issues

N/A

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

yes.

### CODE REVIEW

**Comment by TorstenDittmann** [comment]
`functions.executions.update` is probably the most important one for realtime 🤔

This must probably be triggered from the functions worker :+1:

**Comment by kodumbeats** [comment]
> `functions.executions.update` is probably the most important one for realtime 
> 
> This must probably be triggered from the functions worker

So I understand, you'd like the functions worker to emit the `functions.executions.update` event when a function's status flips (from working to completed, for instance)?

**Comment by TorstenDittmann** [comment]
Yes, this way we can listen to execution completion over realtime ☺️

File: app/config/events.php [hunk 1]
```diff
@@ -97,6 +97,46 @@
...
+        'model' => Response::MODEL_FUNCTION,
+        'note' => 'version >= 0.7',
+    ],
+    'functions.tags.delete' => [
+        'description' => 'This event triggers when a function tag is deleted.',
+        'model' => Response::MODEL_ANY,
+        'note' => 'version >= 0.7',
+    ],
+    'functions.executions.create' => [
+        'description' => 'This event triggers when a function execution is created.',
```
Comment (TorstenDittmann) [inline app/config/events.php:131]: Does the event only trigger when executed over the HTTP endpoint or also on schedule/event?

If only HTTP, I would update this section. Also we are sending update events to any execution, but only send events to specific execution creation? 🤔

Comment (TorstenDittmann) [reply app/config/events.php:131]: Okay, just found the part where its being taken care of 👍🏻

**Review by TorstenDittmann — APPROVED** [review]
lgtm 👍🏻 awesome work @kodumbeats 👏🏻

**Comment by eldadfux** [comment]
Great work @kodumbeats !!

**Pull request merged**